packages/medusa/src/loaders/strategies.ts

```diff
@@ -19,19 +21,40 @@ export default ({ container, configModule, isTest }: LoaderOptions): void => {
     typeof isTest !== "undefined" ? isTest : process.env.NODE_ENV === "test"
 
   const corePath = useMock
-    ? "../strategies/__mocks__/*.js"
-    : "../strategies/*.js"
+    ? "../strategies/__mocks__/[!__]*.js"
+    : "../strategies/**/[!__]*.js"
+
   const coreFull = path.join(__dirname, corePath)
 
-  const core = glob.sync(coreFull, { cwd: __dirname })
+  const core = glob.sync(coreFull, {
+    cwd: __dirname,
+    ignore: ["**/__fixtures__/**", "index.js", "index.ts"],
+  })
+
   core.forEach((fn) => {
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const loaded = require(fn).default
     const name = formatRegistrationName(fn)
-    container.register({
-      [name]: asFunction(
-        (cradle) => new loaded(cradle, configModule)
-      ).singleton(),
-    })
+
+    if (isBatchJobStrategy(loaded.prototype)) {
+      container.registerAdd(
+        "batchJobStrategies",
+        asFunction((cradle) => new loaded(cradle, configModule))
+      )
+
+      container.register({
+        [name]: asFunction(
+          (cradle) => new loaded(cradle, configModule)
+        ).singleton(),
+        [`batch_${loaded.identifier}`]: aliasTo(name),
+        [`batchType_${loaded.batchType}`]: aliasTo(name),
```

Comment:
thought: The naming is a bit inconsistent to me; mixing camelCase with snake_case. Optimally, we'd stick with one, but you might have had a reason to do so?